msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/DiscussionForumTool.java

```diff
@@ -4759,14 +4840,23 @@ public String processDfMsgRevisedPost()
 		// optionally include the revision history. by default, revision history is included
 		boolean showRevisionHistory = ServerConfigurationService.getBoolean("msgcntr.forums.showRevisionHistory",true);
 		if (showRevisionHistory) {
-		    String revisedInfo = "<p class=\"lastRevise textPanelFooter\">" + getResourceBundleString(LAST_REVISE_BY);
-	        revisedInfo += getUserNameOrEid();
-	        revisedInfo  += " " + getResourceBundleString(LAST_REVISE_ON);
-	        SimpleDateFormat formatter = new SimpleDateFormat(getResourceBundleString("date_format"), getUserLocale());
-	        formatter.setTimeZone(getUserTimeZone());
-	        Date now = new Date();
-	        revisedInfo += formatter.format(now) + " </p> ";
-	        currentBody = revisedInfo.concat(currentBody);
+			String revisedInfo = "<p class=\"lastRevise textPanelFooter\">";
+			if (dfTopic.getPostAnonymous())
+			{
+				// Don't reveal display ID or anon ID (this reveals who the editor is). Instead, simply say "Edited on <date>"
+				revisedInfo += getResourceBundleString(LAST_REVISE_ON_ANON) + " ";
+			}
+			else
+			{
+				revisedInfo += getResourceBundleString(LAST_REVISE_BY);
+				revisedInfo += getUserNameOrEid();
+				revisedInfo += " " + getResourceBundleString(LAST_REVISE_ON);
```

Comment:
You could condense this String concatenation into one line; not a big deal though.